src/tools/pub/package-layout.md

```diff
@@ -476,4 +473,46 @@ documentation generators, or other bits of automation.
 Unlike the scripts in `bin`, these are *not* for external users of the package.
 If you have any of these, place them in a directory called `tool`.
 
+## Project specific caching for tools
+
+{{site.alert.info}}
+  Do not check the `.dart_tool/` directory into source control.
+  Instead keep `.dart_tool/` in `.gitignore`.
+{{site.alert.end}}
+
+The `.dart_tool/` directory is created when you run `dart pub get`.
+The directory is used by various tools for caching files specific to your project and/or local machine.
+The `.dart_tool/` directory should never be checked into source control, or copied between machines.
+
+It is also generally safe to delete the `.dart_tool/` directory, though some tools might need recompute the cached information before they work again. 
```

Comment:
Should we strengthen this, so people using the `.dart_tool` directory know what the contract is: "Your files might get deleted at any time".

It is safe to delete the `.dart_tool/` directory, though some tools might need recompute the cached information before they work again.

Reply:
```suggestion
It is safe to delete the `.dart_tool/` directory,
but some tools might need recompute the cached files
before they work again. 
```

Reply:
I believe this is addressed.